### ISSUE TITLE
Find an easy way to use `.at` and `Object.hasOwn`

### ISSUE THREAD

**Issue by fisker** [prettier/prettier#13452]
They are available on Node.js 16, but esbuild won't polyfill for Node.js 14, we do similar thing in `main` branch by using `esbuild -> babel -> esbuild`. But this will slow down build process, and the core-js polyfill seems big.

```sh
npx esbuild core-js/modules/es.array.at.js --bundle
```

It's ~1200 lines, we don't it so prefect. There should be a better way.

**Comment by dnalborczyk** [comment]
https://www.npmjs.com/package/object.hasown
https://www.npmjs.com/package/array.prototype.at

install as global:

```js
import 'object.hasown/auto'

Object.hasOwn({foo: 'bar'}, 'foo') // true
```

**Comment by fisker** [comment]
I wa hoping we can transform `foo.at(-1)` to `at(foo, -1)` and we definite a simple "shim"

```js
const at = (object, index) => object[index < 0 ? object.length + index : index]
```

**Comment by sosukesuzuki** [comment]
Both `object.hasown` and `array.prototype.at` are too big.

**Comment by fisker** [comment]
> install as global:

Also, there is no "global", we have many entries, we need inject to all of them. https://unpkg.com/browse/prettier@2.7.1/

**Comment by fisker** [comment]
I'll try use babel to transform.

**Comment by sosukesuzuki** [comment]
> I'll try use babel to transform.

How will you use Babel? As you said it will slow down build process?

**Comment by fisker** [comment]
I want  to try use babel api in our replaceModule plugin. Not sure how slow can it be.

**Comment by fisker** [comment]
It seems the solution is not too bad #13482 All about 20s in this PR and `next` branch, maybe because our source files are all small. :smile:

**Comment by fisker** [comment]
Fixed in #13482
